### PR TITLE
UI: Improve simple mode replay buffer settings

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Instead of a checkbox in the recording group that reveals the replay
buffer group when pressed, the replay buffer group now has its own
toggle that enables or disables the replay buffer.

Before:
image
image

After:
image
image

### Motivation and Context

There currently is an element of surprise which should be avoided. When clicking the "Enable replay buffer" checkbox, as a visually unrelated group pops up below the recording group, or gets hidden again.
With this change, it's clear on first sight what the checkbox does and what the user can expect when clicking it.

### How Has This Been Tested?

macOS 13
Enabled/Disabled replay buffer, and enabled/disabled lossless quality (and opened the settings window in those states).

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by gxalpha** [comment]
It was discussed off-thread that this is generally fine, but buries the warnings which is undesirable. The warnings should rather be floating on the bottom, like so (Red = ScrollView, Blue = Field for warnings):
image
Converting to draft until that is a thing.

**Comment by gxalpha** [comment]
Waits for #7709

**Comment by gxalpha** [comment]
With https://github.com/obsproject/obs-studio/pull/9338 merged, I rebased this on master so it should be ready for review.
Needs @Warchamp7 approval.

**Review by Warchamp7 — APPROVED** [review]

File: UI/forms/OBSBasicSettings.ui [hunk 1]
```diff
@@ -7855,7 +7851,6 @@
   <tabstop>simpleOutRecFormat</tabstop>
   <tabstop>simpleOutRecEncoder</tabstop>
   <tabstop>simpleOutMuxCustom</tabstop>
-  <tabstop>simpleReplayBuf</tabstop>
```
Comment (WizardCM) [inline UI/forms/OBSBasicSettings.ui:7858]: Missing tabstop, so users might not be able to keyboard-navigate to the settings?

Comment (gxalpha) [reply UI/forms/OBSBasicSettings.ui]: You're right, initially this was removed when the group box kept its name (when I then only changed later).
Undone.

**Pull request merged**